### PR TITLE
Port to Vert.x 4 and sdk-java 2.0.0-milestone2

### LINKED ISSUE #136: Upgrade to cloud events sdk milestone2
**Problem**
We need to update to CloudEvents SDK milestone 2, major changes that impact us are related to the Vert.x 4 support.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
- System Integrator

**Exit Criteria**
Upgrade to cloud events sdk milestone2 and all tests pass.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1

**Additional context (optional)**
- Release notes: https://github.com/cloudevents/sdk-java/releases/tag/2.0.0-milestone2

/area data-plane

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #136

## Proposed Changes

- Update to cloudevents sdk-java 2.0.0-milestone2
- Update to Vert.x 4

**Release Note**

```release-note
Update to Vert.x 4.0.Beta1
Update to cloudevents/sdk-java 2.0.0-milestone2
```

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
The code looks much cleaner than before!

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java [hunk 1]
```diff
@@ -81,12 +82,12 @@ public static void main(final String[] args) {
...
 
     final ConsumerRecordOffsetStrategyFactory<String, CloudEvent>
-        consumerRecordOffsetStrategyFactory = ConsumerRecordOffsetStrategyFactory.create();
+        consumerRecordOffsetStrategyFactory = ConsumerRecordOffsetStrategyFactory.unordered();
 
     final var consumerVerticleFactory = new HttpConsumerVerticleFactory(
         consumerRecordOffsetStrategyFactory,
         consumerConfigs,
-        vertx.createHttpClient(),
+        WebClient.create(vertx, new WebClientOptions().setIdleTimeout(10000)),
```
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:90]: Can we do these kind of things in other prs? I merely removed this hardcoded thing from another point and put there :smile:

**Review by pierDipi — APPROVED** [review]
Opened an issue: https://github.com/knative-sandbox/eventing-kafka-broker/issues/141

Thank you! 

/lgtm

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:90]: Sure: https://github.com/knative-sandbox/eventing-kafka-broker/issues/141

**Pull request merged**